imgcreate/live.py

```diff
@@ -1030,14 +1030,186 @@ def _get_excluded_packages(self):
+	linux /LiveOS/vmlinuz%(index)s root=%(rootlabel)s %(liveargs)s %(extra)s
+	initrd /LiveOS/initrd%(index)s.img
+}
+""" %args
+
+
+    def __get_efi_image_stanzas(self, isodir, name):
+        # FIXME: this only supports one kernel right now...
+        kernels = self._get_kernel_versions()
+        kernel_options = self._get_kernel_options()
+        checkisomd5 = self._has_checkisomd5()
+
+        cfg = ""
+        index = "0"
+        for kernel, version in ((k,v) for k in kernels for v in kernels[k]):
+            self.__copy_kernel_and_initramfs(isodir, version, index)
+            cfg += self.__get_efi_image_stanza(fslabel = self.fslabel,
+                                               liveargs = kernel_options,
+                                               long = "Start " + self.product,
+                                               extra = "", index = index)
+            if checkisomd5:
+                cfg += self.__get_efi_image_stanza(fslabel = self.fslabel,
+                                                   liveargs = kernel_options,
+                                                   long = "Test this media & start " + self.product,
+                                                   extra = "rd.live.check",
+                                                   index = index)
+            cfg += """
+submenu 'Troubleshooting -->' {
+"""
+            cfg += self.__get_efi_image_stanza(fslabel = self.fslabel,
+                                               liveargs = kernel_options,
+                                               long = "Start " + self.product + " in basic graphics mode",
+                                               extra = "nomodeset", index = index)
+
+            cfg+= """}
+"""
+            break
+
+        return cfg
+
+    def _configure_efi_bootloader(self, isodir):
+        """Set up the configuration for an EFI bootloader"""
+        if self.__copy_efi_files(isodir):
+            shutil.rmtree(isodir + "/EFI")
+            logging.warning("Failed to copy EFI files, no EFI Support will be included.")
+            return
+
+        cfg = self.__get_basic_efi_config(isolabel = self.fslabel,
+                                          timeout = self._timeout)
+        cfg += self.__get_efi_image_stanzas(isodir, self.name)
+
+        cfgf = open(isodir + "/EFI/BOOT/grub.cfg", "w")
+        cfgf.write(cfg)
+        cfgf.close()
+
+    def _configure_bootloader(self, isodir):
+        self._configure_efi_bootloader(isodir)
+
 arch = dnf.rpm.basearch(hawkey.detect_arch())
 if arch in ("i386", "x86_64"):
     LiveImageCreator = x86LiveImageCreator
 elif arch in ("ppc",):
     LiveImageCreator = ppcLiveImageCreator
 elif arch in ("ppc64",):
     LiveImageCreator = ppc64LiveImageCreator
-elif arch.startswith(("arm", "aarch64")):
+elif arch.startswith(("aarch64")):
+    LiveImageCreator = aarch64LiveImageCreator
+elif arch.startswith(("arm")):
     LiveImageCreator = LiveImageCreatorBase
```

Comment:
What's the effect of setting `LiveImageCreator` to `LiveImageCreatorBase` ? By looking at that class it seems that image won't work for these arches. Do you want to add an `armLiveImageCreator` that's similar to `aarch64LiveImageCreator` ? Maybe as mentioned you could just create a generic `efiLiveImageCreator` or something and just set `self.efiarch`.

Reply:
Setting it to `LiveImageCreatorBase` means it doesn't work and needs to be implemented.